src/client_side.h

```diff
@@ -162,6 +162,8 @@ class ConnStateData : public Server, public HttpControlMsgSink, private Independ
     virtual void noteBodyConsumerAborted(BodyPipe::Pointer) = 0;
 
     bool handleRequestBodyData();
+    // called to get client connmark
+    const Comm::ConnectionPointer getClientConnection();
```

Comment:
clientConnection is a public member from ::Server. No need to extend the ConnStateData API.